presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -961,7 +1018,8 @@ private Table prepareTable(ConnectorSession session, ConnectorTableMetadata tabl
         Map<String, HiveColumnHandle> columnHandlesByName = Maps.uniqueIndex(columnHandles, HiveColumnHandle::getName);
         List<Column> partitionColumns = partitionedBy.stream()
                 .map(columnHandlesByName::get)
-                .map(column -> new Column(column.getName(), column.getHiveType(), column.getComment()))
+                // TODO: propagate type metadata
```

Comment:
we don't care too much about partition columns. Actually a better could be adding the following method to the `Column` class. Of course, once we add user-defined type info to the column handle, we need to assert here the typeMetadata is empty.

```java
    public static Column partitionColumn(String name, HiveType type, Optional<String> comment)
    {
        return new Column(name, type, comment, Optional.empty());
    }